presentation/src/main/java/me/androidbox/presentation/navigation/NavigationGraph.kt

```diff
@@ -83,7 +93,89 @@ fun NavigationGraph(
                 agendaScreenState = agendaScreenState,
                 agendaScreenEvent = { agendaScreenEvent ->
                     agendaViewModel.onAgendaScreenEvent(agendaScreenEvent)
-                })
+                },
+            onSelectedAgendaItem = {
+                /* TODO The item in the dropdown menu should be an enum or a sealed class that will determine which item was clicked */
+                navHostController.navigate(Screen.EventScreen.route)
+            })
+        }
+
+        /* Event Detail Screen */
+        composable(
+            route = Screen.EventScreen.route
+        ) {
+            val eventViewModel: EventViewModel = hiltViewModel()
+            val eventScreenState by eventViewModel.eventScreenState.collectAsStateWithLifecycle()
+            val title = it.savedStateHandle.get<String>(ContentType.TITLE.name) ?: "New Event"
+            val description = it.savedStateHandle.get<String>(ContentType.DESCRIPTION.name) ?: "New Description"
+
+            LaunchedEffect(key1 = title, key2 = description) {
```

Comment:
I personally did this on the screen, mostly to have all the related logic in one place

But shouldnt be an issue to have it here

Reply:
@MKiperszmid I trying to keep to the pattern of have the Screens ViewModels free for consistency. And I also prefer to pass in the state rather than the ViewModel now.